File: 2k24/01 September/02-Different Ways to Add Parentheses/code.py
class Solution:
    def diffWaysToCompute(self, expression: str) -> List[int]:
        ans=[]
        n=len(expression)
        for i in range(0, n):
            if(expression[i]=='+' or expression[i]=='-' or expression[i]=='*'):
                left_val=self.diffWaysToCompute(expression[:i])
                right_val=self.diffWaysToCompute(expression[i+1:])
                c=expression[i]
                for j in range(0, len(left_val)):
                    for k in range(0, len(right_val)):
                        if(c=='+'):
                            ans.append(left_val[j]+right_val[k])
                        elif(c=='-'):
                            ans.append(left_val[j]-right_val[k])
                        else:
                            ans.append(left_val[j]*right_val[k])
        if(len(ans)==0):
            ans.append(int(expression))
        return ans                                
        

File: 2k24/01 September/02-Different Ways to Add Parentheses/code.cpp
class Solution {
public:
    vector<int> diffWaysToCompute(string expression) {
        vector<int> ans;
        int n=expression.size();
        for(int i=0;i<n;i++){
            if(expression[i]=='+' || expression[i]=='-' || expression[i]=='*'){
                vector<int> left_val=diffWaysToCompute(expression.substr(0, i));
                vector<int> right_val=diffWaysToCompute(expression.substr(i+1));
                char c=expression[i];
                for(int j=0;j<left_val.size();j++){
                    for(int k=0;k<right_val.size();k++){
                        if(c=='+'){ans.push_back(left_val[j]+right_val[k]);}
                        else if(c=='-'){ans.push_back(left_val[j]-right_val[k]);}
                        if(c=='*'){ans.push_back(left_val[j]*right_val[k]);}

                    }
                }

            
            }
        }
        if(ans.size()==0){ans.push_back(stoi(expression));}

        return ans;
    }
};